modules/quest/src/main/java/unnamed/mmo/quest/objective/QuestObjective.java

```diff
@@ -0,0 +1,16 @@
+package unnamed.mmo.quest.objective;
+
+import net.minestom.server.entity.LivingEntity;
+import net.minestom.server.instance.block.Block;
+import unnamed.mmo.quest.QuestProgress;
+
+public interface QuestObjective {
+
+    default void onMobKill(LivingEntity entity) {}
+
+    default void onBlockBreak(Block block) {}
+
+    default void onProgressComplete(QuestProgress progress) {}
+
+    boolean isObjectiveComplete();
```

Comment:
I would also say that quest completion should not be handled in a polling fashion (eg calling something every tick). The vast vast vast majority of ticks have no quest state change, instead a "notification" scheme seems more appropriate. completablefutures or perhaps events come to mind